pkg/secrets/secrets.go

```diff
@@ -81,19 +91,34 @@ func GenerateUserData(publicKey ssh.PublicKey) (*core.Secret, error) {
 			$passwordConf | Set-Content -Path C:\ProgramData\ssh\sshd_config
 			$authorizedKeyFilePath = "$env:ProgramData\ssh\administrators_authorized_keys"
 			New-Item -Force $authorizedKeyFilePath
-			echo "` + string(pubKeyBytes[:]) + `"| Out-File $authorizedKeyFilePath -Encoding ascii
+			echo "` + pubKey + `"| Out-File $authorizedKeyFilePath -Encoding ascii
 			$acl = Get-Acl C:\ProgramData\ssh\administrators_authorized_keys
 			$acl.SetAccessRuleProtection($true, $false)
 			$administratorsRule = New-Object system.security.accesscontrol.filesystemaccessrule("Administrators","FullControl","Allow")
 			$systemRule = New-Object system.security.accesscontrol.filesystemaccessrule("SYSTEM","FullControl","Allow")
 			$acl.SetAccessRule($administratorsRule)
 			$acl.SetAccessRule($systemRule)
 			$acl | Set-Acl
-			Restart-Service sshd
-			</powershell>
-			<persist>true</persist>`),
-		},
-	}
+			Restart-Service sshd`
+}
 
-	return userDataSecret, nil
+// applyTag surrounds the given data using the tag name with the following
+// pattern: <tagName>data</tagName>
+func applyTag(tagName, data string) string {
+	return fmt.Sprintf("<%s>%s</%s>\n", tagName, data, tagName)
+}
```

Comment:
Theres no need to re-engineer this logic, its already available [here](https://github.com/openshift/machine-api-operator/blob/f7f326323f37ab91ec79430769635fc859f15b96/pkg/util/windows/helpers.go#L36)

Reply:
I don't feel that would be appropriate. a) WMCO owns the `windows-user-data` secret, so it's WMCO's responsibility to set the userData based on the platform type and hold the logic for it, b) as per @elmiko's [comment](https://coreos.slack.com/archives/CM4ERHBJS/p1664302470931509?thread_ts=1664290564.001939&cid=CM4ERHBJS) the MAPI logic will be corrected after persist tags are removed (this PR)  and c) [AddPowershellTags](https://github.com/openshift/machine-api-operator/blob/f7f326323f37ab91ec79430769635fc859f15b96/pkg/util/windows/helpers.go#L36) always appends the persist tags, which may become an issue since it is tightly couple with the powershell tags.

The logic proposed in this PR allows a more granular configuration of the required tags per platform type.

Reply:
Makes sense to me